src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/StateSynchronizationPerformanceWindow.cs

```diff
@@ -0,0 +1,104 @@
+            appIPAddress = PlayerPrefs.GetString(appIPAddressKey, "localhost");
+        }
+
+        protected override void OnDisable()
+        {
+            base.OnDisable();
+            PlayerPrefs.SetString(appIPAddressKey, appIPAddress);
+        }
+
+        private void OnGUI()
+        {
+            EditorGUILayout.BeginVertical();
+            {
+                if (GUILayout.Button(new GUIContent("Open performance settings prefab", "Opens the prefab that defines global performance settings."), GUILayout.Width(globalSettingsButtonWidth)))
+                {
+                    StateSynchronizationMenuItems.EditGlobalPerformanceParameters();
+                }
+
+                if (!EditorApplication.isPlaying)
+                {
+                    var observer = FindObjectOfType<StateSynchronizationObserver>();
+                    if (observer == null)
+                    {
+                        RenderTitle("StateSynchronizationObserver was not detected in the current scene. Open the SpectatorViewCompositor scene.", Color.red);
+                    }
+                    else
+                    {
+                        RenderTitle("Enter playmode to view performance information.", Color.gray);
+                    }
+                }
+                else if (EditorApplication.isPlaying &&
+                    !StateSynchronizationObserver.IsInitialized)
+                {
+                    RenderTitle("StateSynchronizationObserver was not detected in the current scene. Open SpectatorViewCompositor", Color.red);
+                }
+                else
+                {
+                    UpdatePerformanceInformation();
+                }
+            }
+            EditorGUILayout.EndVertical();
+        }
+
+        private void UpdatePerformanceInformation()
+        {
+            DeviceInfoObserver stateSynchronizationDevice = null;
+            if (StateSynchronizationObserver.IsInitialized)
+            {
+                stateSynchronizationDevice = StateSynchronizationObserver.Instance.GetComponent<DeviceInfoObserver>();
+            }
+
+            HolographicCameraNetworkConnectionGUI(
+                AppDeviceTypeLabel,
+                stateSynchronizationDevice,
+                GetSpatialCoordinateSystemParticipant(stateSynchronizationDevice),
+                showCalibrationStatus: false,
+                showSpatialLocalization: false,
+                ref appIPAddress);
+
+            if (StateSynchronizationObserver.Instance.PerformanceFeatureCount == 0)
+            {
+                RenderTitle("Waiting for performance information...", Color.yellow);
+            }
+            else
+            {
+                RenderTitle("Performance information", Color.green);
+                scrollPosition = EditorGUILayout.BeginScrollView(scrollPosition);
+                byte index = 0;
+                foreach (var time in StateSynchronizationObserver.Instance.AverageTimePerFeature)
```

Comment:
>foreach (var time in StateSynchronizationObserver.Instance.AverageTimePerFeature) [](start = 16, length = 81)

Should this just be a "for" loop since it needs the index too?

Reply:
sounds good.